src/Groups/group_characters.jl

```diff
@@ -78,32 +78,111 @@ This is the type of (ordinary or Brauer) character tables that can delegate
-    GAPTable::GapObj  # the character table object
+    GAPTable::GapObj  # the GAP character table object
     characteristic::Int
-    GAPGroup::GAPGroup    # the underlying group, if any
+    group::Union{GAPGroup, GrpAbFinGen}    # the underlying group, if any
+    isomorphism::Map  # isomorphism from `group` to a group in GAP
 
-    function GAPGroupCharacterTable(G::GAPGroup, tab::GapObj, char::Int)
-      return new(tab, char, G)
+    function GAPGroupCharacterTable(G::Union{GAPGroup, GrpAbFinGen}, tab::GapObj, iso::Map, char::Int)
+      return new(tab, char, G, iso)
     end
 
     function GAPGroupCharacterTable(tab::GapObj, char::Int)
-      #GAPGroup is left undefined
+      #group is left undefined
       return new(tab, char)
     end
 end
 
+# access to field values via functions
+GAPTable(tbl::GAPGroupCharacterTable) = tbl.GAPTable
+group(tbl::GAPGroupCharacterTable) = tbl.group
+characteristic(tbl::GAPGroupCharacterTable) = tbl.characteristic
+
+# backwards compatibility:
+# return `tbl.group` when one accesses `tbl.GAPGroup`
+function Base.getproperty(tbl::GAPGroupCharacterTable, sym::Symbol)
+   sym === :GAPGroup && return tbl.group
+   return getfield(tbl, sym)
+end
+
+
+# If the character table stores an Oscar group `G` then
+# an isomorphism from `G` to a GAP group is stored in the field `isomorphism`.
+# Different kinds of isomorphisms can be stored,
+# depending on the type of the stored group.
+#
+# - For `GAPGroup`, we compute images and preimages by unwrapping and
+#   wrapping the groups elements, respectively.
+# - For `GrpAbFinGen`, we compose this unwrapping/wrapping with
+#   an isomorphism to a `PcGroup` .
+#
+function isomorphism_to_GAP_group(G::GAPGroup)
+    f = function(x) return x.X; end
+    finv = function(x::GAP.Obj) return group_element(G, x); end
+    return MapFromFunc(f, finv, G, G.X)
+end
+
+function isomorphism_to_GAP_group(G::GrpAbFinGen)
+    @req isfinite(G) "the group is not finite"
+    iso = isomorphism(PcGroup, G)
+    C = codomain(iso)
+    @assert C isa GAPGroup
+    f = function(x) return iso(x).X; end
+    finv = function(x::GAP.Obj) return preimage(iso, group_element(C, x)); end
+    return MapFromFunc(f, finv, G, C.X)
+end
+
+function isomorphism_to_GAP_group(tbl::GAPGroupCharacterTable)
+    @req isdefined(tbl, :isomorphism) "character table stores no isomorphism"
+    return tbl.isomorphism
+end
+
+
+# If the character table stores an Oscar group `G` then
+# the columns correspond to conjugacy classes of `G`.
+# The vector `conjugacy_classes(G)` (stored or not stored in `G`) can be
+# independent of the vector of conjugacy classes of the corresponding
+# group on the GAP side.
+# Therefore we store a vector `conjugacy_classes(tbl)` via an attribute,
+# which is compatible with the `GAP.Globals.ConjugacyClasses` value of
+# the underlying GAP character table.
+function conjugacy_classes(tbl::GAPGroupCharacterTable)
+    return get_attribute!(tbl, :conjugacy_classes) do
```

Comment:
Can't we use the `@attr` method here? (I don't know which return type to use from the top of my head, but the implementation here doesn't have a type stable return value either, so...)
```suggestion
@attr SOME_TYPE function conjugacy_classes(tbl::GAPGroupCharacterTable)
```